Git-commit: 953f06e0f061dd683b2526cc596f38cea3547658
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Tue, 22 Dec 2020 02:14:37 +0000
Subject: [PATCH] PCI: Add MCFG quirks for some Hisilicon Chip host controllers
Patch-mainline: Queued in openEuler repo, version 5.10.0-1.0.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

euler inclusion
category: bugfix
bugzilla: 46851
CVE: NA

-------------------------------------------------

The PCIe controller in some Hisilicon Chip is not completely ECAM-compliant.
Part of its PCIe cores do not support ECAM.

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: yangerkun <yangerkun@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/acpi/pci_mcfg.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/acpi/pci_mcfg.c b/drivers/acpi/pci_mcfg.c
index 95f23acd5b80..3f310cfc9e7d 100644
--- a/drivers/acpi/pci_mcfg.c
+++ b/drivers/acpi/pci_mcfg.c
@@ -77,6 +77,10 @@ static struct mcfg_fixup mcfg_quirks[] = {
 	HISI_QUAD_DOM("HIP07   ",  4, &hisi_pcie_ops),
 	HISI_QUAD_DOM("HIP07   ",  8, &hisi_pcie_ops),
 	HISI_QUAD_DOM("HIP07   ", 12, &hisi_pcie_ops),
+	HISI_QUAD_DOM("HIP12   ", 0x20, &hisi_pcie_ops),
+	HISI_QUAD_DOM("HIP12   ", 0x24, &hisi_pcie_ops),
+	HISI_QUAD_DOM("HIP12   ", 0x28, &hisi_pcie_ops),
+	HISI_QUAD_DOM("HIP12   ", 0x2c, &hisi_pcie_ops),
 
 #define THUNDER_PEM_RES(addr, node) \
 	DEFINE_RES_MEM((addr) + ((u64) (node) << 44), 0x39 * SZ_16M)
-- 
2.31.1

